### PR TITLE
[kubernetes][docs] Implement landing page and getting started guide

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

This PR 

- Implements a landing page for the new KubeRay-based deployment guide.
- Implements a "Getting started" Jupyter notebook with a KinD example that explains how to
  * Deploy the operator and a Ray cluster
  * Use the autoscaler
  * View the dashboard
  * Use Job submission
  * Use Ray client
  * Clean resources up

The purpose of the Getting started page is to provide an interactive overview of the key Ray-on-K8s features.

The location and naming of the new pages might change after https://github.com/ray-project/ray/pull/26754 is merged. The main thing to read for right now is content.

Check "details" on the "docs/readthedocs.org:ray" item of the checks to preview the docs.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kuberay.md
Comment (pcmoritz) [inline doc/source/cluster/kuberay.md:5]: This sounds a bit academic, how about
```
You can execute your distributed Ray programs on a Kubernetes cluster.
```

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [inline doc/source/cluster/kuberay/quickstart.ipynb:12]: `In this guide, we show you how to manage`

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [inline doc/source/cluster/kuberay/quickstart.ipynb:19]: This is only needed if the user is using the Ray client, right? If yes let's shift it to that section, otherwise it might be confusing (since the Ray version is determined from the K8s manifests and not the locally installed Ray version).

Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:19]: You need for job submission too. I think we might as well have people install Ray.

File: doc/source/cluster/kuberay.md
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay.md:5]: Thanks, good call.

Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay.md]: Fixed.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [reply doc/source/cluster/kuberay/quickstart.ipynb:19]: Then maybe add a sentence at the beginning that this is for using the Ray Client and Job submission or similar?

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [inline doc/source/cluster/kuberay/quickstart.ipynb:74]: For kind, is this the case by default?

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [inline doc/source/cluster/kuberay/quickstart.ipynb:92]: Hmm, should this be `kuberay/ray-operator/config/default`? Or otherwise a `cd` needs to be added?

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (pcmoritz) [inline doc/source/cluster/kuberay/quickstart.ipynb:123]: This seems like an advanced use case and should probably not be part of the quickstart? Or have it further down if it is important :)

Comment (pcmoritz) [reply doc/source/cluster/kuberay/quickstart.ipynb:123]: It could also be put into an "advanced" box in sphinx

**Review by pcmoritz — APPROVED** [review]
Thanks, this is great! I made some small comments that should be addressed before merging :)

Overall this feels a little bit more advanced than maybe we want from a quickstart (e.g. mentioning K8s services etc) but I don't have a better suggestion at this point. We can finish the rest of the docs first and then come back to this and try to make it more accessible.

Directionally, it should be possible to go through this doc and have a running Ray cluster on K8s even if people don't know much about K8s.

In particular, K8s shouldn't be the center of the discussion, the center should be how to run Ray applications. K8s is a tool we use to do that.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:123]: This should be shifted into a different page and linked from here. On the other hand, that's already what's done here.
I think later we can shift this into another place in the Ray docs and spell out the details there.  

It _is_ an issue many people encounter when working in a permissions-limited environment and something that's explicitly spelled out in the docs for the old operator.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:92]: Probably, yes.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:74]: Depends on your machine and the local docker setup.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:19]: Will do.

**Comment by DmitriGekhtman** [comment]
Re: K8s vs. Ray focus, I somewhat disagree and think some amount of K8s focus is necessary and unavoidable here.

> In particular, K8s shouldn't be the center of the discussion, the center should be how to run Ray applications. K8s is a tool we use to do that.

This portion of the documentation is on how to use the K8s tool to provision a Ray cluster. There's actually very little here that's specific to Ray -- these docs are pretty much going through the flow for creating a deployment that exposes some services. If you're a K8s novice unfamiliar with the K8s concepts, then you learn them in the process here (because you have to if you want to deploy Ray on K8s and do something nontrivial).

> Directionally, it should be possible to go through this doc and have a running Ray cluster on K8s even if people don't know much about K8s.

This doc achieves that. It also provides a bit of context that will be necessary if you want to do something with the running Ray cluster on K8s.

> e.g. mentioning K8s services

For this bit, it's possible to simply port-forward the head pod. When I implement the "Networking" section, I'll change this guide to do that and shift the service discussion there.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb]: Fixed.

File: doc/source/cluster/kuberay/quickstart.ipynb
Comment (DmitriGekhtman) [reply doc/source/cluster/kuberay/quickstart.ipynb:123]: Keeping it here, for now.

**Review by shrekris-anyscale — APPROVED** [review]
Stamping for codeowner approval.

**Comment by DmitriGekhtman** [comment]
oh good the doc lint finally passed, will merge this in the morning

**Pull request merged**
